slpv2.rflx

```diff
@@ -0,0 +1,289 @@
+   type SLPv2_Flag_Reserved is range 0 .. 0 with Size => 13;
+   type SLPv2 is
+      message
+         Version : SLPv2_Version;
+         Function_ID : Function_ID;
+         Length : SLPv2_Length;
+         Flag_O : Boolean;
+         Flag_F : Boolean;
+         Flag_R : Boolean;
+         Flag_Reserved : SLPv2_Flag_Reserved;
+         Next_Extension_Offset : U24;
+         XID : U16;
+         Language_Tag_Length : U16;
+         Language_Tag : Opaque with Size => Language_Tag_Length * 8
+            then Payload
+               with Size => 8 * Length - (Language_Tag'Last - Version'First + 1)
+               if (8 * Length >= (Language_Tag'Last - Version'First + 1))
+                  and (Next_Extension_Offset = 0)
+            then Payload
+               with Size => 8 * Next_Extension_Offset - (Language_Tag'Last - Version'First + 1)
+               if (Next_Extension_Offset > 0)
+                  and (Next_Extension_Offset < Length)
+                  and (8 * Next_Extension_Offset >= (Language_Tag'Last - Version'First + 1));
+         Payload : Opaque
+            then null
+               if (Next_Extension_Offset = 0)
+                  or (Next_Extension_Offset >= Length)
+            then Extension
+               if (Next_Extension_Offset > 0)
+                  and (Next_Extension_Offset < Length);
+         Extension : Opaque with Size => (Length - Next_Extension_Offset) * 8;
+      end message;
+
+   -- 8.1. Service Request
+   type Service_Request is
+      message
+         Length_Of_PR_List : U16;
+         PR_List_String : Opaque with Size => Length_Of_PR_List * 8;
+         Length_Of_Service_Type : U16;
+         Service_Type_String : Opaque with Size => Length_Of_Service_Type * 8;
+         Length_Of_Scope_List : U16;
+         Scope_List_String : Opaque with Size => Length_Of_Scope_List * 8;
+         Length_Of_Predicate_String : U16;
+         Service_Request_Predicate : Opaque with Size => Length_Of_Predicate_String * 8;
+         Length_Of_SLP_SPI_String : U16;
+         SLP_SPI_String : Opaque with Size => Length_Of_SLP_SPI_String * 8;
+      end message;
+   for SLPv2 use (Payload => Service_Request)
+      if Function_ID = FI_SERVICE_REQUEST;
+
+   -- 9.2. Authentication Blocks
+   type Authentication_Block_Authentication_Block_Length is range 10 .. 2**16-1 with Size => 16;
+   type Authentication_Block is
+      message
+         Block_Structure_Descriptor : U16;
+         Authentication_Block_Length : Authentication_Block_Authentication_Block_Length;
+         Authentication_Block_Timestamp : U32;
+         SLP_SPI_String_Length : U16;
+         SLP_SPI_String : Opaque with Size => SLP_SPI_String_Length * 8;
+         Structured_Authentication_Block : Opaque
+            with Size => 8 * Authentication_Block_Length - (SLP_SPI_String'Last - Block_Structure_Descriptor'First + 1)
+               if (8 * Authentication_Block_Length >= (SLP_SPI_String'Last - Block_Structure_Descriptor'First + 1));
```

Comment:
The condition (`if`) should have the same indentation as the size aspect (`with`).

Reply:
Fixed.